### PR TITLE
GraphQL Request Orchestrator and About Account in API

### PR DESCRIPTION
**Pull request by dangeredwolf**
> [!NOTE]
> Introduces a reusable GraphQL orchestration layer and enriches user responses with AboutAccount metadata.
> 
> - **New orchestrator**: `graphql/orchestrator.ts` executes weighted/fallback GraphQL methods and parallel requests with per-method validators
> - **Status fetching refactor**: `conversation.ts` now uses `graphQLOrchestrator` for single-status retrieval with dynamic weights (prioritizes `TweetDetail` when threading) and simplified fallback/404 handling
> - **User API enhancements**: `profile.ts` fetches `UserByScreenName`/`UserResultByScreenName` via orchestrator and runs `AboutAccountQuery` in parallel; merges `about_account` (based_in, location accuracy, created-country accuracy, source, username change history) into `APIUser`
> - **Types**: Extend `APIUser.verification` with optional `identity_verified` and add `about_account` fields; adjust vendor `AboutAccountQueryResponse`
> - **Tests**: Replace `env-wrapper` with `helpers/harness`; add mocks and tests for `AboutAccountQuery`; split user tests into `api.user.test.ts`; update existing tests to use harness
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 8685f22b87b538547ac50ea9647c1503d2ed675b. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

## Summary by CodeRabbit

* **New Features**
  * Enhanced user profiles now include AboutAccount details (based_in, location accuracy, creation-country accuracy, source, username change history).
  * Added identity verification flag to profile verification data.

* **Refactor**
  * Consolidated and improved tweet and profile fetching for more consistent, reliable retrieval and error handling.

* **Tests**
  * Added tests and mocks covering AboutAccount fetching, validation, and edge cases.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: src/providers/twitter/conversation.ts [hunk 1]
```diff
@@ -408,89 +408,113 @@ const filterBucketStatuses = (tweets: GraphQLTwitterStatus[], original: GraphQLT
...
+              | GraphQLTwitterStatus
+              | TweetStub
+              | undefined;
+            return Boolean((r as GraphQLTwitterStatus)?.__typename || (r as TweetStub)?.reason);
+          }
+        },
+        {
+          name: 'TweetResultsByRestIds',
+          query: TweetResultsByRestIdsQuery,
+          weight: processThread ? 0 : 500,
```
[the remark this answers is not in the document]
Comment (dangeredwolf) [reply src/providers/twitter/conversation.ts]: all good now as of https://github.com/FxEmbed/FxEmbed/commit/eb02a6f4040d7fd22f57619d50dd2d7203db85cd

File: test/helpers/harness.ts [hunk 2]
```diff
@@ -91,6 +91,17 @@ export default {
...
+          const screenNameAbout = variables.screenName;
+          // load mock based on screen name
+          try {
+            const mock = await import(`../mocks/AboutAccountQuery/${screenNameAbout}.json`);
+            console.log('Mock data:', mock);
+            return new Response(JSON.stringify(mock));
+          } catch (error) {
+            console.error('Error loading mock:', error);
+            return new Response(JSON.stringify({ data: {} }));
+          }
```
[the remark this answers is not in the document]
Comment (dangeredwolf) [reply test/helpers/harness.ts:104]: But it shouldn't cause behavioral differences since it will _always_ return something... right?

File: test/profile.test.ts [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+    envWrapper
+  );
+
+  expect(result.status).toEqual(200);
+  const response = (await result.json()) as UserAPIResponse;
+  const user = response.user as APIUser;
+
+  // If username_changes exists, count should be a number >= 0
+  expect(user.about_account?.username_changes?.count).toBeGreaterThanOrEqual(0);
+});
```
[the remark this answers is not in the document]
Comment (dangeredwolf) [reply test/profile.test.ts]: removed irrelevant tests

File: test/profile.test.ts [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+  expect(result.status).toEqual(200);
+  const response = (await result.json()) as UserAPIResponse;
+  const user = response.user as APIUser;
+
+  // If username_changes exists, last_changed_at should be string or null
+  expect(
+    user.about_account?.username_changes?.last_changed_at === null ||
+      typeof user.about_account?.username_changes?.last_changed_at === 'string'
+  ).toBe(true);
+});
```
[the remark this answers is not in the document]
Comment (dangeredwolf) [reply test/profile.test.ts]: removed irrelevant tests

File: test/profile.test.ts [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+  const user = response.user as APIUser;
+  expect(user).toBeTruthy();
+
+  // User should still be returned even if AboutAccountQuery fails
+  expect(user.id).toBeTruthy();
+  expect(user.screen_name).toBeTruthy();
+
+  // about_account may be undefined if AboutAccountQuery failed
+  // This is acceptable - the user data should still be present
+});
```
[the remark this answers is not in the document]
Comment (dangeredwolf) [reply test/profile.test.ts]: removed irrelevant tests

File: test/helpers/harness.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (dangeredwolf) [reply test/helpers/harness.ts:104]: It's just a test harness to ensure it pulls offline data instead of trying to load from the internet during tests, so it's touched very infrequently anyway
